app/uk/gov/hmrc/securemessage/services/SecureMessageService.scala

```diff
@@ -121,22 +127,50 @@ class SecureMessageService @Inject()(
     }
   }
 
-  def addMessageToConversation(
+  def addCaseWorkerMessageToConversation(
+    client: String,
+    conversationId: String,
+    messagesRequest: CaseworkerMessageRequest)(
+    implicit ec: ExecutionContext,
+    hc: HeaderCarrier): Future[Either[SecureMessageError, Unit]] =
+    repo.conversationExists(client, conversationId).flatMap { exists =>
```

Comment:
i would move logic to repo.getConversation(clientId: String, conversationId): Future[Either[ConversationIdNotFound, Conversation]]
Then in the service you just add another line the the for comprehension :)